platform/darwin/src/NSPredicate+MGLAdditions.mm

```diff
@@ -0,0 +1,11 @@
+#import "NSPredicate+MGLAdditions.h"
+
+@implementation NSPredicate (MGLAdditions)
+
+- (mbgl::style::Filter)mgl_filter
+{
+    [NSException raise:@"A specific predicate must override this method." format:@""];
```

Comment:
The raise parameter should be a short title. The format parameter should be a complete sentence describing the problem.